src/commonMain/kotlin/com/akelius/university/statistic/core/ScoreCalculator.kt

```diff
@@ -30,75 +29,65 @@ import kotlin.math.abs
  *
  *  Interactions will be recognized if they have weight>0 and score>0.
  *
+ *  4) In slideshow with one slide score is implemented in Interaction itself.
+ *  It defaults because of it to default formula
+ *
  */
 class ScoreCalculator {
 
     companion object {
         const val calculationPrecision = 0.00000001
-        val gradeCalculator = GradeCalculator()
+        internal val gradeCalculator = GradeCalculator()
     }
 
     fun calculate(score: SlideshowScore): SlideshowScoreResult {
         val calculation = Calculation()
 
-        if (score.slideScores.isNotEmpty() && score.slideScores.find { it.weight > 0 } == null) {
-            // if there is not a single weight bigger than zero, this is presentation where we give always 5 points
-            return SlideshowScoreResult(
-                score = 5,
-                scaledScore = 1.0,
-                correctAnswersCount = score.slideScores.size,
-                totalAnswersCount = score.slideScores.size
-            )
-        }
-
         score.slideScores.asSequence().forEach {
             calculation.addSlideScore(it)
         }
+
+        val countInteractions = score.slideScores.count { it.weight > 0 }
+        val calculationType = calculationFormula(score, countInteractions)
+
         val scaledScore = calculateScaledScore(calculation.userScore, calculation.maxScore)
 
-        when (score.slideScores.count { it.weight > 0 }) {
-            1 -> {
-                val scoreInFifths = gradeCalculator.scoreToFifths(score.type, scaledScore)
-                val interaction = score.slideScores.find { it.weight > 0 }
-                return SlideshowScoreResult(
-                    score = scoreInFifths,
-                    scaledScore = scaledScore,
-                    correctAnswersCount = if(interaction!!.isCorrect) 1 else 0,
-                    totalAnswersCount = 1
-                )
-            }
-            in 2..19 -> {
-                val mistakes = calculateMistakes(score)
-
-                val scoreInFifths = gradeCalculator.scoreToFifths(CalculationType.MISTAKE, mistakes.toDouble())
-                return SlideshowScoreResult(
-                    score = scoreInFifths,
-                    scaledScore = scaledScore,
-                    correctAnswersCount = calculation.correctAnswersCount,
-                    totalAnswersCount = calculation.totalAnswers
-                )
-            }
-            else -> {
-                val scoreInFifths = gradeCalculator.scoreToFifths(score.type, scaledScore)
-                return SlideshowScoreResult(
-                    score = scoreInFifths,
-                    scaledScore = scaledScore,
-                    correctAnswersCount = calculation.correctAnswersCount,
-                    totalAnswersCount = calculation.totalAnswers
-                )
+        val scoreInFifths = gradeCalculator.scoreToFifths(
+            calculationType, if (calculationType == CalculationType.MISTAKE) {
```

Comment:
I'd extract this "if" statement into a method `getScore` or with smth similar name - because it looks awful to me